### PR TITLE
OLS-466: make console UI deployment replica configurable

### PR DESCRIPTION
**Pull request by raptorsun**
## Description

The console UI deployment uses the the field `ols.spec.deployment.replicas` as the appserver deployment does.
No new fields is added to CRD.

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes # [OLS-466](https://issues.redhat.com//browse/OLS-466)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [x] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: internal/controller/ols_console_ui_assets.go [hunk 1]
```diff
@@ -112,7 +111,7 @@ func (r *OLSConfigReconciler) generateConsoleUIDeployment(cr *olsv1alpha1.OLSCon
 			Labels:    generateConsoleUILabels(),
 		},
 		Spec: appsv1.DeploymentSpec{
-			Replicas: &replicas,
+			Replicas: cr.Spec.OLSConfig.DeploymentConfig.Replicas,
```
Comment (xrajesh) [inline internal/controller/ols_console_ui_assets.go:114]: @raptorsun Does this make `one replica` count for all our deployments ? We will not be able to control like 1 service Pod , 3 console pod etc ?

Comment (raptorsun) [reply internal/controller/ols_console_ui_assets.go:114]: In that case, we can add a new field to the console config structure.

Comment (raptorsun) [reply internal/controller/ols_console_ui_assets.go:114]: let me post a new version with a separate replica count for console

**Comment by raptorsun** [comment]
/override "Red Hat Konflux / ols-enterprise-contract / bundle"
/override "Red Hat Konflux / ols-enterprise-contract / lightspeed-operator"
/override "Red Hat Konflux / ols-enterprise-contract / test-bundle"
/override "Red Hat Konflux / ols-operator-e2e-tests-415 / test-bundle"
/override "Red Hat Konflux / ols-operator-e2e-tests-416 / test-bundle"
/override "Red Hat Konflux / ols-operator-e2e-tests-417 / test-bundle"

**Pull request merged**